interpreter/k8s/src/main/scala/io/buoyant/interpreter/k8s/ConfigMapInterpreterInitializer.scala

```diff
@@ -41,7 +42,7 @@ case class ConfigMapInterpreterConfig(
   private[this] val log = Logger()
 
   @JsonIgnore
-  val api = {
+  def api = {
```

Comment:
Doesn't this create a new client for every time `api` is referenced?

Reply:
Yes, but `api` is only referenced once, and leaving this as a `val` causes [the test interpreter](https://github.com/linkerd/linkerd/pull/1640/commits/4c3b885c60e56a8e67b6d3b7e5e458587597d301#diff-fe9d877a01176ddad3635ce17d9e22c7R163) to throw a `NullPointerException`.

Reply:
Scala: where the only thing more insane than the problems are the solutions.

```
object TestConfigMapInterpreterConfig extends {
  override val api = v1.Api(service).withNamespace("test")
} with ConfigMapInterpreterConfig(
  None, None, Some("test"), "test-config", "test.dtab"
) {
```

Reply:
Jesus Christ. That fixes it???? I didn't even know you _could_ extend `{ .. }`, I thought you could only do `new ConfigMapInterpreter(...) { ... }`

Reply:
These little corners of the language are so disgusting yet fascinating.